File: unitri-ppi/aulas-spring/src/main/java/com/example/aulasspring/controller/PessoaController.java

package com.example.aulasspring.controller;

import com.example.aulasspring.model.Pessoa;
import com.example.aulasspring.service.PessoaService;
import org.springframework.beans.factory.annotation.Autowired;
import org.springframework.stereotype.Controller;
import org.springframework.validation.BindingResult;
import org.springframework.web.bind.annotation.GetMapping;
import org.springframework.web.bind.annotation.PathVariable;
import org.springframework.web.bind.annotation.PostMapping;
import org.springframework.web.bind.annotation.RequestMapping;
import org.springframework.web.servlet.ModelAndView;

import javax.validation.Valid;


@Controller
@RequestMapping(value = "/pessoas")
public class PessoaController {

    @Autowired
    private PessoaService service;

    @GetMapping
    public ModelAndView findAll() {

        ModelAndView mv = new ModelAndView("/pessoas");
        mv.addObject("pessoas", service.findAll());

        return mv;
    }

    @GetMapping(value = "/add")
    public ModelAndView add(Pessoa pessoa) {

        ModelAndView mv = new ModelAndView("/pessoa");
        mv.addObject("pessoa", pessoa);

        return mv;
    }

    @GetMapping(value = "/edit/{id}")
    public ModelAndView edit(@PathVariable("id") Long id) {

        return add(service.findOne(id));
    }

    @GetMapping(value = "/delete/{id}")
    public ModelAndView delete(@PathVariable("id") Long id) {

        service.delete(id);
        return findAll();
    }

    @PostMapping(value = "/save")
    public ModelAndView save(@Valid Pessoa pessoa, BindingResult result) {

        if (result.hasErrors()) {
            return add(pessoa);
        }
        service.save(pessoa);
        return findAll();
    }
}

File: unitri-ppi/aula-07/src/main/kotlin/com/example/aula07/controller/PessoaController.kt
package com.example.aula07.controller

import com.example.aula07.model.Pessoa
import com.example.aula07.service.PessoaService
import org.springframework.stereotype.Controller
import org.springframework.validation.BindingResult
import org.springframework.web.bind.annotation.GetMapping
import org.springframework.web.bind.annotation.PathVariable
import org.springframework.web.bind.annotation.PostMapping
import org.springframework.web.bind.annotation.RequestMapping
import org.springframework.web.servlet.ModelAndView
import javax.validation.Valid


@Controller
@RequestMapping("/pessoas")
class PessoaController(
    val service: PessoaService
) {

    @GetMapping
    fun findAll(): ModelAndView =
        ModelAndView("/pessoas").addObject("pessoas", service.findAll())

    @GetMapping("/add")
    fun add(pessoa: Pessoa): ModelAndView = ModelAndView("/pessoa").addObject("pessoa", pessoa)

    @GetMapping("/edit/{id}")
    fun edit(@PathVariable("id") id: Long): ModelAndView = add(service.findOne(id))

    @GetMapping("/delete/{id}")
    fun delete(@PathVariable("id") id: Long): ModelAndView {
        service.delete(id)
        return findAll()
    }

    @PostMapping("/save")
    fun save(@Valid pessoa: Pessoa, result: BindingResult): ModelAndView {

        if (result.hasErrors()) {
            return add(pessoa)
        }
        service.save(pessoa)
        return findAll()
    }

}